### PR TITLE
feat: cf worker for listing snapshot files

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- Make `https://forest-archive.chainsafe.dev/list` available.

**Reference issue to close (if applicable)**

Closes

**Other information and links**

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
@lemmih let's add latest snapshot listing as well so folks can get easily static links and drive this to `main`

File: cf/forest-snapshot-listing/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# http://editorconfig.org
```
Comment (LesnyRumcajs) [inline cf/forest-snapshot-listing/.editorconfig:1]: What is this? Is this enforced in CI in any way?

File: cf/forest-snapshot-listing/src/worker.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/**
+ * Welcome to Cloudflare Workers! This is your first worker.
+ *
+ * - Run "npm run dev" in your terminal to start a development server
+ * - Open a browser tab at http://localhost:8787/ to see your worker in action
+ * - Run "npm run deploy" to publish your worker
+ *
+ * Learn more at https://developers.cloudflare.com/workers/
+ */
+
```
Comment (LesnyRumcajs) [inline cf/forest-snapshot-listing/src/worker.ts:10]: Not first anymore!
```suggestion
```

File: cf/forest-snapshot-listing/src/worker.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		switch (pathname) {
+			case '/list/': {
+				const html = `<!DOCTYPE html>
+		      <body>
+		      <h1>Forest Archive</h1>
+          <ul>
+          <li><a href="/list/calibnet/diff">Calibnet Diffs</a></li>
+          <li><a href="/list/calibnet/lite">Calibnet Lite</a></li>
+          <li><a href="/list/calibnet/lite">Calibnet Latest</a></li>
```
Comment (LesnyRumcajs) [inline cf/forest-snapshot-listing/src/worker.ts:65]:
```suggestion
          <li><a href="/list/calibnet/latest">Calibnet Latest</a></li>
```

File: cf/forest-snapshot-listing/src/worker.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+          <li><a href="/list/mainnet/diff">Mainnet Diffs</a></li>
+          <li><a href="/list/mainnet/lite">Mainnet Lite</a></li>
+          <li><a href="/list/mainnet/lite">Mainnet Latest</a></li>
```
Comment (LesnyRumcajs) [inline cf/forest-snapshot-listing/src/worker.ts:68]:
```suggestion
          <li><a href="/list/mainnet/latest">Mainnet Latest</a></li>
```

**Review by LesnyRumcajs — COMMENTED** [review]
Would you like to add a README here?

**Comment by LesnyRumcajs** [comment]
Can you run it in `--dev`?

File: cf/forest-snapshot-listing/.editorconfig (hunk 1 above)
Comment (lemmih) [reply cf/forest-snapshot-listing/.editorconfig:1]: It was auto-created. Not sure who reads it.

**Comment by lemmih** [comment]
> Can you run it in `--dev`?

`wranger dev --remote` works for me.

File: cf/forest-snapshot-listing/.editorconfig (hunk 1 above)
Comment (lemmih) [reply cf/forest-snapshot-listing/.editorconfig]: Deleted the file.

File: cf/forest-snapshot-listing/src/worker.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	});
+}
+
+export default {
+	async fetch(request, env) {
+		const url = new URL(request.url);
+		const { pathname } = url;
+
+		switch (pathname) {
+			case '/list/': {
```
Comment (LesnyRumcajs) [inline cf/forest-snapshot-listing/src/worker.ts:48]: make it more user-who-forgots-slashes-friendly
```suggestion
			case '/list':
			case '/list/': {
```

File: cf/forest-snapshot-listing/src/worker.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+	return new Response(html, {
+		headers: {
+			'content-type': 'text/html;charset=UTF-8',
+		},
+	});
+}
+
+export default {
+	async fetch(request, env) {
```
Comment (LesnyRumcajs) [inline cf/forest-snapshot-listing/src/worker.ts:43]: shall we allow only `GET`? Not a big deal, though.

**Review by LesnyRumcajs — APPROVED** [review]

File: cf/forest-snapshot-listing/src/worker.ts (hunk 6 above)
Comment (lemmih) [reply cf/forest-snapshot-listing/src/worker.ts:43]: Done. Limited to GET requests.

**Comment by ruseinov** [comment]
will check this out asap.

File: cf/forest-snapshot-listing/src/worker.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  html += `</ul></body>`;
+
+  return new Response(html, {
+    headers: {
+      'content-type': 'text/html;charset=UTF-8',
+    },
+  });
+}
+
+export default {
```
Comment (ruseinov) [inline cf/forest-snapshot-listing/src/worker.ts:46]: nit: I would rather use a template as opposed to generate html within the controller code, but if we aren't going to touch this much - I guess it does not make a whole lot of difference.

**Review by ruseinov — APPROVED** [review]

**Pull request merged**